src/haddock/libs/libio.py

```diff
@@ -113,6 +118,32 @@ def read_from_yaml(yaml_file: FilePath) -> dict[Any, Any]:
     return ycfg
 
 
+def check_yaml_duplicated_parameters(yaml_fpath: str) -> None:
+    """Make sure the provided yaml file do not contain duplicated parameters.
+
+    Parameters
+    ----------
+    yaml_fpath : str
+        Path to a yaml file
+    """
+    # Build regular expression
+    yaml_param_regex = re.compile("^(([A-Za-z0-9]_?)+):")
```

Comment:
Regexes get confusing very fast for someone else looking into the code. It's best if you could generate an example with https://regex101.com and paste it here